open-ce/test_feedstock.py

```diff
@@ -0,0 +1,223 @@
+
+    Args:
+        name (str): The name of the test that was run.
+        returncode (int): The return code of the test that was run.
+        output (str): The resuling output from running the test.
+    """
+    def __init__(self, name, returncode, output):
+        self.name = name
+        self.output = output
+        self.returncode = returncode
+
+    def display_failed(self):
+        """
+        Display the output from a failed test.
+        """
+        if self.failed():
+            print("--------------------------------")
+            print("Failed test: " + self.name)
+            print(self.output)
+            print("--------------------------------")
+
+    def failed(self):
+        """
+        Returns whether or not a test failed.
+        """
+        return self.returncode != 0
+
+def load_test_file(test_file):
+    """
+    Load a given test file.
+
+    Args:
+        test_file (str): Path to the test file to load.
+    """
+    with open(test_file, 'r') as stream:
+        test_file_data = yaml.safe_load(stream)
+
+    return test_file_data
+
+def gen_test_commands(test_file=utils.DEFAULT_TEST_CONFIG_FILE):
+    """
+    Generate a list of test commands from the provided test file.
+
+    Args:
+        test_file (str): Path to the test file.
+    """
+    test_data = load_test_file(test_file)
+
+    time_stamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+    conda_env = "open-ce-conda-env-" + time_stamp
+
+    test_commands = []
+
+    # Create conda environment for testing
+    test_commands.append(TestCommand(name="Create conda environment " + conda_env,
+                                     conda_env=conda_env,
+                                     create_env=True))
+
+    for test in test_data['tests']:
+        test_commands.append(TestCommand(name=test.get('name'), conda_env=conda_env, bash_command=test.get('command')))
+
+    test_commands.append(TestCommand(name="Remove conda environment " + conda_env,
+                                     conda_env=conda_env,
+                                     clean_env=True))
+
+    return test_commands
+
+def run_test_commands(conda_env_file, test_commands):
+    """
+    Run a list of tests within a conda environment.
+
+    Args:
+        conda_env_file (str): The name of the conda environment file used to create the conda environment.
+        test_commands (:obj:`list` of :obj:`TestCommand): List of test commands to run.
+    """
+    failed_tests = []
+    for test_command in test_commands:
+        test_result = test_command.run(conda_env_file)
+        if test_result.failed():
+            failed_tests.append(test_result)
+
+    return failed_tests
+
+def display_failed_tests(failed_tests):
+    """
+    Display a list of failed tests.
+
+    Args:
+        failed_tests (:obj:`list` of :obj:`TestResult`): A list of failed TestResult's.
+    """
+    # If any collected output, return 1, else return 0
+    if failed_tests:
```

Comment:
We probably want failed test lists and full outputs to be written to disk somewhere as well.

Reply:
I will open an issue for that. I think there are some existing file formats that we could output the logs in that would allow other tools to parse them better.

For now, a user can just pipe the output to a file if they are interesting in storing the output of the tests.

Reply:
Ok, sounds good.